xarray/testing/duckarrays.py

```diff
@@ -0,0 +1,140 @@
+from abc import abstractmethod
+from typing import TYPE_CHECKING
+
+import hypothesis.extra.numpy as npst
+import hypothesis.strategies as st
+import numpy as np
+import numpy.testing as npt
+import pytest
+from hypothesis import given, note
+
+import xarray as xr
+import xarray.testing.strategies as xrst
+from xarray.core.types import T_DuckArray
+from xarray.testing.assertions import assert_identical
+
+if TYPE_CHECKING:
+    from xarray.core.types import _DTypeLikeNested, _ShapeLike
+
+
+__all__ = [
+    "VariableConstructorTests",
+    "VariableReduceTests",
+]
+
+
+class ArrayConstructorChecksMixin:
+    """Mixin for checking results of Variable/DataArray constructors."""
+
+    def check(self, var, arr):
+        self.check_types(var, arr)
+        self.check_values(var, arr)
+        self.check_attributes(var, arr)
+
+    def check_types(self, var, arr):
+        # test type of wrapped array
+        assert isinstance(
+            var.data, type(arr)
+        ), f"found {type(var.data)}, expected {type(arr)}"
+
+    def check_attributes(self, var, arr):
+        # test ndarray attributes are exposed correctly
+        assert var.ndim == arr.ndim
+        assert var.shape == arr.shape
+        assert var.dtype == arr.dtype
+        assert var.size == arr.size
+
+    def check_values(self, var, arr):
+        # test coercion to numpy
+        npt.assert_equal(var.to_numpy(), np.asarray(arr))
+
+
+class VariableConstructorTests(ArrayConstructorChecksMixin):
+    shapes = npst.array_shapes()
+    dtypes = xrst.supported_dtypes()
+
+    @staticmethod
+    @abstractmethod
+    def array_strategy_fn(
+        *, shape: "_ShapeLike", dtype: "_DTypeLikeNested"
+    ) -> st.SearchStrategy[T_DuckArray]:
+        # TODO can we just make this an attribute?
+        ...
+
+    @given(st.data())
+    def test_construct(self, data) -> None:
+        shape = data.draw(self.shapes)
+        dtype = data.draw(self.dtypes)
+        arr = data.draw(self.array_strategy_fn(shape=shape, dtype=dtype))
```

Comment:
You can't use `strategies.variables` to test the constructor because `strategies.variables` has a call to the constructor inside it, so you end up not being able to access the original raw array that you want to use as the "expected" value.